homeassistant/components/shelly/__init__.py

```diff
@@ -155,27 +156,14 @@ async def async_services_setup(
     async def async_service_ota_update(call: ServiceCall):
         """Trigger OTA update."""
         if not (call.data.get(ATTR_DEVICE_ID) or call.data.get(ATTR_AREA_ID)):
-            _LOGGER.warning("OTA update service: no device or area selected")
+            _LOGGER.warning("OTA update service: no target selected")
             return
 
-        devices = []
-        if call.data.get(ATTR_AREA_ID):
-            for area_id in call.data.get(ATTR_AREA_ID):
-                devices += [
-                    area_dev
-                    for area_dev in device_registry.async_entries_for_area(
-                        dev_reg, area_id
-                    )
-                    if DOMAIN in next(iter(area_dev.identifiers))
-                ]
-
-        if call.data.get(ATTR_DEVICE_ID):
-            for device_id in call.data.get(ATTR_DEVICE_ID):
-                device = dev_reg.async_get(device_id)
-                if not any(device.id == x.id for x in devices):
-                    devices += [device]
-
-        for device in devices:
+        devices = await async_extract_referenced_entity_ids(hass, call)
```

Comment:
This return value is a dataclass with different attributes that return entity_ids, device_ids and area_ids.

```suggestion
        selected_ids = await async_extract_referenced_entity_ids(hass, call)
```

Reply:
Thanks for that proposal